drivers/kscan/kscan_cap1203.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2022 Keiya Nobuta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_cap1203
+
+#include <drivers/kscan.h>
+#include <drivers/i2c.h>
+#include <drivers/gpio.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(cap1203, CONFIG_KSCAN_LOG_LEVEL);
+
+#define REG_MAIN_CONTROL 0x0
+#define CONTROL_INT 0x1
+
+#define REG_INPUT_STATUS 0x03
+
+#define REG_INTERRUPT_ENABLE 0x27
+#define INTERRUPT_ENABLE 0x7
+#define INTERRUPT_DISABLE 0x0
+
+struct cap1203_config {
+	struct i2c_dt_spec i2c;
+	struct gpio_dt_spec int_gpio;
+};
+
+struct cap1203_data {
+	struct device *dev;
+	kscan_callback_t callback;
+	struct k_work work;
+	/* Interrupt GPIO callback. */
+	struct gpio_callback int_gpio_cb;
+#ifdef CONFIG_KSCAN_CAP1203_POLL
+	/* Timer (polling mode). */
+	struct k_timer timer;
+#endif
+};
+
+static int cap1203_clear_interrupt(const struct i2c_dt_spec *i2c)
+{
+	uint8_t ctrl;
+	int r;
+
+	r = i2c_reg_read_byte_dt(i2c, REG_MAIN_CONTROL, &ctrl);
+	if (r < 0) {
+		return r;
+	}
+
+	ctrl = ctrl & ~CONTROL_INT;
+	return i2c_reg_write_byte_dt(i2c, REG_MAIN_CONTROL, ctrl);
+}
+
+static int cap1203_enable_interrupt(const struct i2c_dt_spec *i2c, uint8_t enable)
```

Comment:
please use booleans for boolean flags
```suggestion
static int cap1203_enable_interrupt(const struct i2c_dt_spec *i2c, bool enable)
```

Reply:
The value to be written to the register is specified here, but I think that boolean is more appropriate, so I'll change it.